### PR TITLE
commit failing tests to demonstrate bugs;

### PR DESCRIPTION
**Pull request by taharrison**
what to do when master is broken;
etc.

### CODE REVIEW

File: CardinalRules.md [hunk 1]
```diff
@@ -172,17 +172,20 @@ Fixed bug = test x N
...
-Your code in each commit should compile and pass unit tests, no broken tests should be committed.
+We call a commit **broken** if it fails to compile, fails to pass all tests, or fails to deploy successfully.
+
+**Never (knowingly) merge broken code** to the develop or master branches, or to release branches.
 
-At an absolute minimum, you should commit your code before you go home to your local repository and to origin as well.
+However, you can commit broken code to feature or hotfix branches if it is a work in progress and it will be fixed before any further commits are made to that branch.  In particular it may be very useful to write and commit a failing test to highlight a bug or a missing feature.  Further commits to a branch with failing tests should be efforts toward fixing the failing test.
 
-Be smart – commit frequently.
+If broken code is merged to develop, release, or master (note that build servers may run more tests - e.g. integration tests, UI tests - than local environments), you should immediately tag the commit, then on the server **reset the branch head to the previous commit** - this should be rare!
```
Comment (taharrison) [inline CardinalRules.md]: note, resetting branch head makes sense.  talk to me and alan.

**Comment by westonized** [comment]
LGTM

**Comment by Gonzales** [comment]
LGTM

**Comment by stevemchugh** [comment]
This should link to instructions (another md file) for how to reset the head in this scenario. Doesn't have to be in the StyleGuide necessarily, but if we are prescribing an action then we should be able to see how to go about doing it.

**Comment by petefitz** [comment]
This is better. Does this also cover when a push is made to origin\master when it shouldn't have ? It was necessary to reset the head on the server at that time, and ensure offending local was also reverted to avoid the same being pushed again.

**Comment by petefitz** [comment]
Otherwise, LGTM

**Comment by taharrison** [comment]
@petefitz - linking this to issue https://github.com/burrowsyr/Styleguide/issues/15

**Pull request merged**